components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -159,6 +179,51 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) t
 
     }
 
+    /**
+     * Validate the signature algorithm according to FAPI specification.
+     * According to FAPI, signature algorithm should be PS256 or ES256.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations">...</a>
+     *
+     * @param clientId  client id
+     * @param algorithm signature algorithm
+     * @return is valid signature algorithm
+     */
+    private static boolean isValidFAPISignatureAlgorithm(String clientId, String algorithm) {
+
+        List<String> allowedFAPIAlgorithms = new ArrayList<>();
```

Comment:
As the FAPI allowed algorithms are added as a config lets get from there instead of hardcoded values.

ref: https://github.com/wso2-extensions/identity-oauth-addons/pull/108/files#diff-c1fefc87f2cf0485b2f5e352367c515cb1036459a5232698108ff48d4d50a5e1R187-R188

Reply:
fixed with [404b385](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2156/commits/404b385d6065754da691e95faaac308c2b9d298a)